bindings/go/ctf/filesystem_ctf_test.go

```diff
@@ -193,3 +195,76 @@ func Test_FileSystemCTF_FileSystemOperations(t *testing.T) {
 	_, err = os.Stat(filepath.Join(tmpDir, testFile))
 	r.NoError(err)
 }
+
+type mockBlob struct {
+	blob.ReadOnlyBlob
+	digest string
+	known  bool
+	size   int64
+	reader io.ReadCloser
+	err    error
+}
+
+func (m *mockBlob) Digest() (string, bool) {
+	return m.digest, m.known
+}
+
+func (m *mockBlob) Size() int64 {
+	return m.size
+}
+
+func (m *mockBlob) ReadCloser() (io.ReadCloser, error) {
+	if m.err != nil {
+		return nil, m.err
+	}
+	return m.reader, nil
+}
+
+func TestFileSystemCTF_SaveBlob_ErrorCases(t *testing.T) {
+	tests := []struct {
+		name    string
+		blob    *mockBlob
+		wantErr string
+	}{
+		{
+			name: "blob not digest aware",
+			blob: &mockBlob{
+				digest: "",
+				known:  false,
+			},
+			wantErr: "blob does not have a digest that can be used to save it",
+		},
+		{
+			name: "digest not known",
+			blob: &mockBlob{
+				digest: "sha256:123",
+				known:  false,
+			},
+			wantErr: "blob does not have a digest that can be used to save it",
```

Comment:
Is such a setup even possible? It sounds counterintuitive to have the digest filled, but the `known` field `false`.

Reply:
consider it a sanity check if you use a blob implementation that doesnt expose a digest!